### PR TITLE
Add syntax for escaped field selectors.

### PR DESCRIPTION
**Pull request by jnthntatum**
Introduces syntax for escaped fields that are not normally syntactically legal field selections with backticks.

This slightly relaxes the parse requirements for fields to cover more types of field like keys e.g. 

```
tags.`skewer-case-name` 
proto.`com.example.ext.field_name`
paths.`/api/v1`
```

This PR is limited to fields, but we could consider generalizing to cover namespaced names and simple identifier for variables in a later update (e.g. `in.example.Foo` or `package`)

### CODE REVIEW

**Review by TristonianJones — COMMENTED** [review]
I like this as a first approach at relaxing the syntax to support escapes need for fields named after reserved words and for supporting proto extension fields natively. 

The type-checker should continue to validate field names which is great, and there's no weirdness with invalid function names, just escaping on the namespaces. It seems to help satisfy a common pain point. It might not solve top-level .in domains within namespace names, but it's a great start.

File: test/proto2pb/test_all_types.proto [hunk 1]
```diff
@@ -112,7 +112,7 @@ message NestedTestAllTypes {
 // with fully qualified names.
 message ExampleType {
   optional string name = 1;
-
+  optional int64 in = 2;
```
Comment (TristonianJones) [inline test/proto2pb/test_all_types.proto:115]: Can you make sure to update the canonical test protos with the same field name in a PR on cel-spec?

File: ext/protos_test.go [hunk 2]
```diff
@@ -42,12 +42,21 @@ func TestProtos(t *testing.T) {
 		{expr: `proto.getExt(ExampleType{}, google.expr.proto2.test.nested_example) == ExampleType{}`},
 		{expr: `proto.getExt(ExampleType{}, google.expr.proto2.test.ExtendedExampleType.extended_examples) == []`},
 		{expr: `proto.getExt(ExampleType{}, google.expr.proto2.test.ExtendedExampleType.enum_ext) == GlobalEnum.GOO`},
+		{expr: "ExampleType{`in`: 64}.`in` == 64"},
+		// TODO(jnthntatum): legal parse, but can't assign extension fields without updates to type checker and runtime
```
Comment (TristonianJones) [inline ext/protos_test.go]: This is interesting. Are you thinking of doing this in a follow-up PR?

**Review by TristonianJones — DISMISSED** [review]
LGTM. My questions which can be addressed post-submission.

File: test/proto2pb/test_all_types.proto (hunk 1 above)
Comment (jnthntatum) [reply test/proto2pb/test_all_types.proto:115]: Will do.

File: ext/protos_test.go (hunk 2 above)
Comment (jnthntatum) [reply ext/protos_test.go]: Done. I added a reference, can you re-approve?

**Review by TristonianJones — APPROVED** [review]

**Pull request merged**